src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,125 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int BIT_DIVIDER = 1;
+    private static final int SOURCE_POSITION = 0;
+    private static final int DESTINATION_POSITION = 0;
+    private static final int SHIFT_INDEX_NUMBER = 1;
+    private static final int SIZE_INITIALIZER = 0;
+    private int size;
+    private Object[] elements;
```

Comment:
You need to use the parametrize type
```suggestion
    private T [] elements;
```